packages/create-svelte/shared/+libskeleton+checkjs/jsconfig.json

```diff
@@ -9,6 +9,6 @@
 		"skipLibCheck": true,
 		"sourceMap": true,
 		"strict": true,
-		"moduleResolution": "NodeNext"
+		"moduleResolution": "bundler"
```

Comment:
Not sure we can change this - for people who _author_ packages and then publish them, the `.js` file endings still are necessary if regular node should be able to resolve it. Then again, this is not the case for Svelte libraries since they are noExternal'd and therefore loaded through bundlers which probably can deal with it - we need to test this.

Reply:
@dummdidumm You're saying consumers should use `moduleResolution": "bundler"` but projects published to NPM should stick with `"moduleResolution": "NodeNext"`?

Reply:
I think so, yes

Reply:
I don't understand the subtleties here but what about packages that are both consumed by others _and_ depend on other packages? Still `"NodeNext"`?

Reply:
Ok, I've reverted that part for now. Maybe we should set it in the generated `tsconfig.json` file? But we'd have to somehow detect which version of TypeScript the user has, so I'm not quite sure how we'd handle it. Maybe we leave that for a separate change

Reply:
> I don't understand the subtleties here but what about packages that are both consumed by others and depend on other packages? Still "NodeNext"?

Yes. As soon as you make something that's consumable by others and which should be loaded through Node, it needs the `.js` file endings because of ESM resolution mode - strictly speaking, at least. It could be that when you have a Svelte library which you can only consume via a bundler anway, that it doesn't matter then, because the bundler knows how to deal with the imports without file endings.